src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,99 @@
+package duke.util;
+
+import duke.exception.DukeException;
+
+import java.util.Scanner;
+
+public class Parser {
+    private final Scanner sc;
+    private final List list;
+
+    public Parser(List list) {
+        this.sc = new Scanner(System.in);
+        this.list = list;
+    }
+
+    // extract int from done and delete prompts
+    public static int extractInt(String input) throws DukeException {
+        String[] parts = input.split(" ");//split along the whitespace to get the integer
+        if (parts.length <= 1) {// checking for incomplete prompts
+            throw new DukeException();
+        }
+        String numStr = parts[1];
+        return Integer.valueOf(numStr);
+    }
```

Comment:
I believe public non-trivial methods like these (as well as other methods in this class and other classes) should be documented by Javadocs. 😅

Reply:
yea updated thanks

Reply:
I think you should come up with a more comprehensive javadoc for this method.
Use the `@param` and `@return` tags to explain what your input is and what you return